### PR TITLE
Add News article

### PR DESCRIPTION
**Pull request by dbosen**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-4.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.
- [ ] New features or changes are documented.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -240,6 +240,7 @@ jobs:
 
           composer update
           composer exec -- drush updb -y
+          composer exec -- drush pm:enable thunder_news_article --yes
```
Comment (chrfritsch) [inline .github/workflows/tests.yml]: Why do we need this? Shouldn't it be enough to enable the module in the test base classes?

File: config/install/simple_sitemap.bundle_settings.article.node.news_article.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+index: true
```
Comment (chrfritsch) [inline config/install/simple_sitemap.bundle_settings.article.node.news_article.yml]: Does the news article need its own sitemap variant?

File: tests/src/FunctionalJavascript/ThunderArticleTestTrait.php [hunk 3]
```diff
@@ -17,20 +17,25 @@ trait ThunderArticleTestTrait {
...
-    $this->drupalGet('node/add/article');
+  public function nodeFillNew(array $fieldValues, string $type): void {
+    $this->drupalGet('node/add/' . $type);
     $this->assertWaitOnAjaxRequest();
+    if (empty($fieldValues)) {
+      return;
+    }
 
-    if (!empty($fieldValues)) {
+    file_put_contents('/tmp/output.html', $this->getSession()->getPage()->getContent());
```
Comment (chrfritsch) [inline tests/src/FunctionalJavascript/ThunderArticleTestTrait.php]:
```suggestion
```

File: modules/thunder_news_article/thunder_news_article.install [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        $role->grantPermission('revert news_article revisions');
+      }
+      if ($role->hasPermission('view article revisions')) {
+        $role->grantPermission('view news_article revisions');
+      }
+      if ($role->hasPermission('access_unpublished node article')) {
+        $role->grantPermission('access_unpublished node news_article');
+      }
+      $role->save();
+    }
```
Comment (chrfritsch) [inline modules/thunder_news_article/thunder_news_article.install]: You can create an array `['has_permission' => 'grant_permission']` out of this

File: modules/thunder_news_article/thunder_news_article.install [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    if (\Drupal::moduleHandler()->moduleExists('simple_sitemap')) {
+      $config_path = Drupal::service('extension.list.profile')->getPath('thunder') . '/config/install';
+      $source = new FileStorage($config_path);
+      $config_storage = \Drupal::service('config.storage');
+      $config_storage->write('simple_sitemap.bundle_settings.article.node.news_article', $source->read('simple_sitemap.bundle_settings.article.node.news_article'));
+    }
```
Comment (chrfritsch) [inline modules/thunder_news_article/thunder_news_article.install]: Why not as optional config?

**Review by chrfritsch — DISMISSED** [review]

File: modules/thunder_news_article/thunder_news_article.install (hunk 4 above)
Comment (dbosen) [reply modules/thunder_news_article/thunder_news_article.install]: doh

File: tests/src/FunctionalJavascript/ChannelsTagsTest.php [hunk 6]
```diff
@@ -35,28 +35,28 @@ public function testChannelsCreation(): void {
...
-    $this->articleFillNew([
+    // Create 2. News Article.
+    $this->nodeFillNew([
       'field_channel' => 6,
       'title[0][value]' => 'Article 2',
       'field_seo_title[0][value]' => 'Article 2',
       'field_tags[]' => [[7, 'New Section'], 'Tag2'],
       'field_teaser_text[0][value]' => 'Teaser 2',
-    ]);
+    ], 'article');
```
Comment (IT-Cru) [inline tests/src/FunctionalJavascript/ChannelsTagsTest.php]: Should this not be news_article here?

**Review by IT-Cru — APPROVED** [review]
Past with some adjustment in usage of project related field tokens for meta tags and LD+JSON structured data our QA.

**Pull request merged**